docs/core/tools/dotnet-build.md

```diff
@@ -163,4 +163,8 @@ Build a project and its dependencies for a specific runtime (in this example, Ub
 
 Build the project and use the specified NuGet package source during the restore operation (.NET Core SDK 2.0 and later versions):
 
-`dotnet build --source c:\packages\mypackages`
\ No newline at end of file
+`dotnet build --source c:\packages\mypackages`
+
+Build the project and set 1.2.3.4 version as a build parameter
+
+`dotnet build /p:Version=1.2.3.4`
```

Comment:
@ryzngard please consider the following comments:
- to be consistent with all other commands, let's use a dash: `dotnet build -p:Version=1.2.3.4`
- `-p` option should be documented; it's not in the synopsis section now, neither in the output of the `dotnet build -h`.

Reply:
Unfortunately that won't work, but we probably should document why. - parameters are interpreted as dotnet command options, which are shown with -h. Anything that's not a - parameter or a build command, after a valid build command is given, are provided as "app arguments". 

It looks like the behavior is intentional. I tried digging into the dotnet cli source to see, but haven't quickly been able to identify if that's true.

Reply:
>Unfortunately that won't work, 

How would it behave then? 

I've just run
>dotnet build -p:Version=1.42.3.4

successfully: the resulted `dll` has the file version "1.42.3.4"
Using .NET Core SDK 2.1.300

Reply:
Hmm... I thought it wouldn't work! Glad to know it does. I just tested on a local machine as well, and you're right 👍 

I'll add some documentation for the -p switch. Maybe see if I can get it included with -h output in CLI as well. The interesting bit is that it doesn't seem to be part of the dotnet cli parsing in https://github.com/dotnet/cli/blob/54bce746005c1ef0bdec0a7f4c560c3dbc0d79a0/src/dotnet/commands/dotnet-build/BuildCommandParser.cs, which is why I thought it wouldn't work with "-"; I believe msbuild requires /p.

Reply:
@ryzngard I don't know if you've noticed (sorry, if  yes), the current page already documents `-p` switch:
>In addition to its options, the dotnet build command accepts MSBuild options, such as -p for setting properties or -l to define a logger. For more information about these options, see the MSBuild Command-Line Reference.

Reply:
Though, it's not in the **Synopsis** section

Reply:
@ryzngard thanks for the link to the `BuildCommandParser`: from it I've learnt that the following:
```
dotnet build -property:Version=1.42.3.4
```

also works (with one dash). Is it worth including to the docs?

Reply:
I did completely miss that! From the current wording, any MSbuild option would be accepted right? Maybe it would be best to include that in **Options** and the **Synopsis** sections to highlight this. Conversely, adding some sort of heading to highlight that MSBuild is used and all command line arguments for MSBuild are accepted would be good? That way lazy people (like myself) can easily identify the information on the screen

Reply:
>also works (with one dash). Is it worth including to the docs?

Yes! Document all the things :)

Reply:
> From the current wording, any MSbuild option would be accepted right? 

At least I understand current wording this way :).

>Maybe it would be best to include that in Options and the Synopsis sections to highlight this.

For example, move/copy the following:
> In addition to its options, the dotnet build command accepts MSBuild options, such as -p for setting properties or -l to define a logger. For more information about these options, see the MSBuild Command-Line Reference.

to the bottom of the **Options** section? To make it more on display.